Cython/Compiler/ModuleNode.py

```diff
@@ -773,7 +782,7 @@ def generate_filename_table(self, code):
                 if isabs(file_path):
                     file_path = basename(file_path)  # never include absolute paths
                 escaped_filename = file_path.replace("\\", "\\\\").replace('"', r'\"')
-                code.putln('"%s",' % escaped_filename)
+                code.putln('%s,' % EncodedString(escaped_filename).as_c_string_literal())
```

Comment:
This will not work for non-UTF8 file names in Py2. Not sure right now if this is a real issue, but it probably is.
Why was this change needed?

Reply:
It needs `as_c_string_literal` to avoid putting unicode characters in the C source. `replace` returns a plain `unicode` object rather than an `EncodedString` even if the input is an `EncodedString` so it does need converting.

It does pass the testsuite in Python2 (the majority of which is plain ASCII filenames) so I think it isn't a real issue.